package/batocera/core/batocera-scripts/scripts/batocera-create-collection

```diff
@@ -0,0 +1,170 @@
+#
+# Bugs? Improvements? Contact 'lbrpdx' on Batocera Discord and forum
+#
+# 20200331 - Initial version
+#
+rompath=/userdata/roms
+collecpath=/userdata/system/configs/emulationstation/collections
+systemscfg=/usr/share/emulationstation/es_systems.cfg
+user_systemscfg=/userdata/system/configs/emulationstation/es_systems.cfg
+
+# Usage 
+print_help() {
+cat << EOF 
+batocera-create-collection [ -c collection_name ] [ -g genre ] search_terms 
+  -c foobar  : The resulting collection will be stored
+               as "custom-foobar.cfg" in the collection directory
+  -g Shooter : Only games matching "Shooter" <genre> in your gamelist.xml 
+               files will be added to the collection
+  -s snes    : Only games in the /userdata/snes/ directory (to filter out
+               systems -- 1 system only after '-s')
+  -h         : Display this help file
+
+Note that existing game collections with the same name will have newly
+found games added to the list.
+
+Examples:
+  "batocera-create-collection zelda" creates a custom collection
+  of all your roms matching "Zelda" in their names.
+
+  You can use it with "final fantasy", and it creates a collection file
+  named "custom-final.fantasy.cfg".
+
+  Invoke "batocera-create-collection -c basketball 'nba '" if you want a list 
+  of NBA games named "custom-basketbell.cfg", without 'pinball' games in.
+
+  "batocera-create-collection -c plateformers -g Plateform" creates a 
+  collection of all plarform games (into custom-plateformers.cfg).
+
+  Finally, "batocear-create-collection -c all" creates a custom-all.cfg 
+  collection with all your games. It can take a while to create, if you have 
+  a lot of games on your system.
+
+  If you want to create a collection with Zelda games for both nes and gb,
+  you can start with "batocera-create-collection -s nes zelda" and then
+  add gb games with "batocera-create-collection -s gb zelda".
+
+Gotchas: 
+  - 'search_terms' for games are case insensitive...
+  - ... but 'genres' and 'systems' are case sensitive (blame scapers disparities)
+
+EOF
+}
+
+# Does a list contain an element?
+contains_element() {
+	local e match="$1"
+	shift
+	for e; do 
+		[[ "$e" == "$match" ]] && return 0
+	done
+	return 1
+}
+
+# Look for ROM suitable extensions
+build_ext_array() {
+	[ -f ${user_systemscfg} ] && systemscfg=${user_systemscfg}
+	all_ext=$(xmllint --xpath '/systemList/system/extension/text()' ${systemscfg} 2>/dev/null)
+	ext_array="" 
+	for i in ${all_ext}; do
+		! contains_element ${i} ${ext_array} && ext_array="$i $ext_array" 
+	done
+	extensions="("$(echo ${ext_array}|sed "s/ /|/g")")"
+}
+
+# First, let's do a match with ROM filenames
+# We'll parse only roms from matching extensions
+parse_filenames() {
+	fn=$(date +"%s")
+	tmpfile=/tmp/collec_${fn}
+	depth=1
+	[[ -z $systempath ]] && depth=2
+	find ${rompath}/${systempath} -type f -maxdepth ${depth} | egrep -i "${rompath}.*${pattern}.*${extensions}" > ${tmpfile}
```

Comment:
```
find ${rompath}/${systempath} -type f -maxdepth ${depth} | egrep -i "[^/]*$.*${pattern}.*${extensions}" > ${tmpfile}
```

Will ignore false positives for because path search will be ignored and only filename matters.

Reply:
That doesn't work with:
```
/userdata/roms/nes/Mega Man (USA).zip
/userdata/roms/nes/Mega Man 4 (USA).zip
/userdata/roms/nes/Mega Man 2 (USA).zip
/userdata/roms/nes/Mega Man 3 (USA).zip
/userdata/roms/nes/Mega Man 5 (USA).zip
/userdata/roms/nes/Mega Man 6 (USA).zip
```
With your command `find /userdata/roms/nes -type f -maxdepth 1 | egrep -i '[^/]*$.*mega.*'` returns nothing

Reply:
@lbrpdx 
This is the correct pattern ... sorry
```
find ${rompath}/${systempath} -type f -maxdepth ${depth} | egrep -i "[^/]*${pattern}.*${extensions}$"
```